docs/class-definitions.md

```diff
@@ -1,247 +1,476 @@
-## Class Definitions
+# Spaces
 
-**Contents**
+## Constructor
 
-- [Spaces](#spaces)
-- [SpaceOptions](#spaceoptions)
-- [Space](#space)
-- [SpaceMember](#spacemember)
-- [Locations](#locations-1)
-- [Cursors](#cursors-1)
-- [Cursor](#cursor)
-- [CursorPosition](#cursorposition)
+Create a new instance of the Spaces library.
 
-### Spaces
+The Spaces library constructor is overloaded allowing it to be instantiated using a [ClientOptions](https://ably.com/docs/api/realtime-sdk?lang=javascript#client-options) object:
 
-#### constructor(ably)
+```ts
```

Comment:
Should we do this alongside the changes to the code that remove this option?

Reply:
I don't think changing the code should be a blocker to updating docs if that is our agreed recommendation.

Reply:
Ideally we should do docs changes alongside code changes like this, otherwise the TypeScript types are out of sync for example (providing the option for different ways to initialise).

Reply:
Ok, fair enough wrt the types then. Can we add a sentence to mention which option we recommend and the fact that the other will be deprecated soon?